Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -92,6 +96,119 @@ namespace gdjs {
           .replace(/[^\w|-]/g, '')
           .slice(0, 30);
       };
+
+      const checkLeaderboardAvailability = async function (
+        url: string
+      ): Promise<boolean> {
+        try {
+          const response = await fetch(url, {
+            method: 'GET',
+            headers: { 'Content-Type': 'application/json' },
+          });
+          if (!response.ok) {
+            logger.error(
+              `Error while fetching leaderboard view, server returned: ${response.status} ${response.statusText}`
+            );
+            return false;
+          }
+          return true;
+        } catch (err) {
+          logger.error(`Error while fetching leaderboard view: ${err}`);
+          return false;
+        }
+      };
+
+      const receiveMessage = function (
+        runtimeScene: gdjs.RuntimeScene,
+        event: MessageEvent
+      ) {
+        if (event.data === 'closeLeaderboardView') {
+          closeLeaderboardView(runtimeScene);
+        }
+      };
+
+      export const displayLeaderboard = async function (
+        runtimeScene: gdjs.RuntimeScene,
+        leaderboardId: string
+      ) {
+        const gameId = gdjs.projectData.properties.projectUuid;
+        const targetUrl = `https://liluo.io/games/${gameId}/leaderboard/${leaderboardId}`;
+        if (!(await checkLeaderboardAvailability(targetUrl))) {
+          logger.error('Leaderboard data could not be fetched, doing nothing');
+          return;
+        }
+
+        if (_leaderboardViewIframe) {
+          _leaderboardViewIframe.src = targetUrl;
+        } else {
+          const domElementContainer = runtimeScene
+            .getGame()
+            .getRenderer()
+            .getDomElementContainer();
+          if (!domElementContainer) {
+            logger.error(
+              "Div covering game couldn't be found, leaderboard cannot be displayed."
+            );
+            return;
+          }
+          const iframe = document.createElement('iframe');
+
+          iframe.src = targetUrl;
+          iframe.id = 'leaderboard-view';
+          iframe.style.position = 'absolute';
+          iframe.style.pointerEvents = 'all';
+          iframe.style.top = '0px';
+          iframe.style.height = '100%';
+          iframe.style.left = '0px';
+          iframe.style.width = '100%';
+          iframe.style.border = 'none';
+          _leaderboardViewIframe = iframe;
+          if (typeof window !== 'undefined') {
+            _leaderboardViewClosingCallback = (event: MessageEvent) => {
+              receiveMessage(runtimeScene, event);
+            };
+            (window as any).addEventListener(
```

Comment:
Is `window as Window` working?

Reply:
No, TS doesn't seem to be able to cast it